### ISSUE TITLE
REST gateway hides error codes

### ISSUE THREAD

**Issue by delatbabel** [thephpleague/omnipay-paypal#120]
The PayPal error "names" are in fact what other gateways refer to as error codes (these can be alphanumeric).  They are documented here:

https://developer.paypal.com/docs/api/#payments-errors

However in PayPal REST the result of getCode() is the HTTP response code, not the PayPal error code.  This makes problem resolution difficult because the actual error codes are hidden by this code.

It would be a non-bc change to provide the actual error name instead of the HTTP response code so I will leave this open for discussion for the time being.

**Comment by kayladnls** [comment]
Based on the docs it sounds like exposing the actual error code returned from the gateway would be ideal. :+1:

**Comment by mantas-done** [comment]
Yes, would be great to have it. 
With PayPal Pro payment, getCode() doesn't return status at least for me
but error message is present in getMessage()

**Comment by delatbabel** [comment]
Yes, but this note is for REST only, not Pro.  In any case I suggest you switch to REST, it is a more recent gateway and better supported (by us and by PayPal). The classic APIs are unlikely to be deprecated soon but it could happen one day.

**Comment by njdubois** [comment]
Are we getting anywhere with this?  Paypal doesn't like something, would be great to know what!
